### ISSUE TITLE
Intelligently wait until previous deletion has completed

### ISSUE THREAD

**Issue by recursionbane** [mrishab/google-photos-delete-tool#50]
The current iteration with timers between deletion cycles is somewhat clunky, and does not take machine performance and network latency into account.

Recommended change is to wait until the "Moving X images" prompt disappears from the DOM before attempting the next deletion cycle.

**Comment by recursionbane** [comment]
I made some other changes like removing the limit of retries in this fork: https://github.com/recursionbane/google-photos-delete-tool

**Comment by mrishab** [comment]
Sounds amazing. Can you make a PR back into the project with this change?

**Comment by mrishab** [comment]
## AI-Generated Summary and Implementation Plan

**Issue Analysis**: This is a smart improvement suggestion to replace fixed sleep/timer delays with intelligent DOM-based waiting. Instead of using `setTimeout()` with a fixed duration, the script should poll the DOM to detect when Google Photos has completed the deletion operation.

**Current Implementation Problem**:
- Fixed timers (e.g., `await sleep(5000)`) don't account for network latency or machine performance
- Creates unnecessary wait times when deletion completes faster
- May timeout on slow connections when deletion takes longer than expected

**Proposed Solution - Intelligent Waiting**:

1. **Create a function to detect deletion completion**:
   ```javascript
   async function waitForDeletionComplete() {
     const maxWaitTime = 60000; // 60 second max timeout
     const pollInterval = 500; // Check every 500ms
     const startTime = Date.now();
     
     while (Date.now() - startTime < maxWaitTime) {
       // Look for the "Moving X images" prompt
       const movingPrompt = document.querySelector('[aria-label*="Moving"]');
       if (!movingPrompt) {
         // Prompt is gone, deletion completed
         return true;
       }
       await sleep(pollInterval);
     }
     // Timeout reached, return false
     return false;
   }
   ```

2. **Replace existing delay logic**:
   - Remove or replace `await sleep(5000)` or similar calls
   - Call `waitForDeletionComplete()` instead
   - This ensures you only proceed when deletion actually completes

3. **Add safety timeout**:
   - Always include a maximum wait time to prevent infinite loops
   - If timeout is reached, log a warning and continue anyway

4. **Error handling**:
   - Gracefully handle cases where the prompt structure changes
   - Add console logging for debugging

**Implementation Steps**:
1. Identify the exact DOM selector/attribute for the "Moving X images" prompt
2. Create the `waitForDeletionComplete()` function
3. Replace all fixed delay timers with this function
4. Test with various network conditions and Google Photos states
5. Add appropriate logging and error handling

**Benefits**:
- ✅ Faster overall execution on good connections
- ✅ Better reliability on slow connections  
- ✅ More robust to machine performance variations
- ✅ Cleaner, more professional implementation

**Status Note**: The requestor has already created a fork with this improvement. Consider reviewing their fork or requesting a PR from them to merge this enhancement.

**Recommendation**: High priority - this is a clear improvement that makes the tool more reliable and efficient.

**Comment by shtse8** [comment]
Agreed — this repo uses fixed timers. The maintained continuation replaces them with polling: it waits until the "Moving X items" dialog appears, confirms, and only then starts the next batch, with timeouts and exponential backoff: https://github.com/shtse8/Google-Photos-Delete-Tool (Chrome extension, userscript, bookmarklet, or console paste).
